File: 1013-fibonacci-number/fibonacci-number.go
func fib(n int) int {
    if n == 0 || n == 1 {
        return n
    }

    n1, n2 := 0, 1

    for i := 2; i <= n; i++ {
        curr := n1 + n2

        n1 = n2
        n2 = curr

    }
    return n2
}

File: 1013-fibonacci-number/fibonacci-number.js
/**
 * @param {number} n
 * @return {number}
 */
var fib = function(n) {

    if(n < 2){
        return n;
    }
    let first_num = 0;
    let second_num = 1;

    for(let i = 2; i <= n; i++) {
        let third_num = first_num + second_num;

        first_num = second_num;
        second_num = third_num;
    }
    return second_num;
};